data/org.freedesktop.portal.Settings.xml

```diff
@@ -73,8 +73,25 @@
       @value: The value @key is set to.
 
       Reads a single value. Returns an error on any unknown namespace or key.
```

Comment:
Maybe document what it actually returns too?

Reply:
I'm not sure what you mean here - document that it's a variant inside another variant? I find it quite hard to work clearly, which is why the deprecation line below suggests it's broken and you should use ReadOne instead.

Reply:
![image](https://github.com/flatpak/xdg-desktop-portal/assets/11528996/2cb0d7e6-0d75-4ed6-8859-fa7d83f4c394)

Correcting the already existing docs would be enough imo, ` OUT v `-> ` OUT (v)`

Reply:
No, `(v)` is a variant inside a (pointless) tuple. The existing `Read` method is already correctly returning a variant: the problem is that if the setting is (for example) a string, instead of the intended result (a variant containing a string), it's returning a variant containing a variant containing a string. In GLib's GVariant notation, instead of `<'hello'>`, it's returning `<<'hello'>>`.

Return type `(v)` would be `(<something>,)`, which is incompatible, and still doesn't give you any more information about what the *something* is.

Reply:
well, it has to be documented anyways that it returns variant(variant(what you actually want))